### DISCUSSION TITLE
Is there a way to get the current URL in a variable?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1529] [Q&A] [answered]
### Question

Hi All,

I am trying to create a email verification link for new account sign ups. So upon sign up, I will send them an email with a link, which they click and I activate their account. 

I am just using the ```on_air``` thing so its a random URL. I have created a page like ```ui.page('\activate')```. So the link will have that part plus a token generated during sign up. When they click the link they come to the ```activate?token=``` page and the code validates the token.

When I use ```on_air```, the URL will be randomly generated, something like ```https://on-air.nicegui.io/devices/D4F136qB/```
So is there a way to get this string in a variable at the back end?

Thanks,
Anindya

**Comment by natankeddem** [comment, marked as answer]
Does this work?
```py
from nicegui import app, ui


async def show_url():
    a = await ui.run_javascript(f'new URL(window.location.href)')
    ui.notify(a)
ui.button('show url', on_click=show_url)
ui.run()
```

**Comment by rodja** [comment]
You could also use 

```py
@ui.page('/')
async def index(client: Client):
    await client.connected()
    url = await ui.run_javascript('window.location.href')
    print(url, flush=True)
```

But in both solutions the url will only be available after the page is accessed remotely. I'll create a pull request to make the URL available programatically after connection to the On Air server is established.

**Comment by rodja** [comment]
I just created #1537 which will allow you to access the urls via `app.urls`.

**Reply by bbence84** [reply]
This would be super awesome! :) I know this feature is beta, but I would send the url via email, which would allow me to open the url without having to log into the Rasperry Pi device on which the nicegui server runs. Thanks!

**Comment by Anindya088** [comment]
Hi @rodja , @natankeddem , 

Thanks so much both of you, I just tested out and it works beautifully. 

Regards,
Anindya
